src/PrestaShopBundle/Service/Form/MultistoreCheckboxAttacher.php

```diff
@@ -0,0 +1,118 @@
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Service\Form;
+
+use PrestaShop\PrestaShop\Core\Domain\Configuration\ShopConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+use PrestaShop\PrestaShop\Core\Feature\FeatureInterface;
+use PrestaShop\PrestaShop\Core\Multistore\MultistoreContextCheckerInterface;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+use Symfony\Component\Form\FormInterface;
+
+class MultistoreCheckboxAttacher
+{
+    private $multistoreFeature;
+    private $configuration;
+    private $context;
+
+    /**
+     * MultistoreCheckboxAttacher constructor.
+     *
+     * @param FeatureInterface $multistoreFeature
+     * @param ShopConfigurationInterface $configuration
+     * @param MultistoreContextCheckerInterface $context
+     */
+    public function __construct(
+        FeatureInterface $multistoreFeature,
+        ShopConfigurationInterface $configuration,
+        MultistoreContextCheckerInterface $context
+    ) {
+        $this->multistoreFeature = $multistoreFeature;
+        $this->configuration = $configuration;
+        $this->context = $context;
+    }
+
+    /**
+     * @return bool
+     */
+    public function shouldAddCheckboxes(): bool
+    {
+        if (!$this->multistoreFeature->isUsed() || $this->context->isAllShopContext()) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Adds multistore checkboxes to form fields if needed,
+     *
+     * @param FormInterface $form (passed by reference)
+     *
+     * @throws \PrestaShop\PrestaShop\Core\Domain\Shop\Exception\ShopException
+     */
+    public function addCheckboxes(FormInterface &$form): void
+    {
+        foreach ($form->all() as $child) {
+            $options = $child->getConfig()->getOptions();
+            if (!isset($options['attr']['multistore_configuration_key'])) {
+                continue;
+            }
+
+            // Check if current configuration is overridden by current shop / group shop context
+            $shopConstraint = new ShopConstraint(
+                $this->context->getContextShopId(),
+                $this->context->getContextShopGroup()->id,
+                true
+            );
+            $isOveriddenInCurrentContext = $this->configuration->has($options['attr']['multistore_configuration_key'], $shopConstraint);
+
+            // update current field with disabled attribute
+            $options['attr']['disabled'] = !$this->context->isAllShopContext() && !$isOveriddenInCurrentContext;
+            $form->add(
+                $child->getName(),
+                get_class($child->getConfig()->getType()->getInnerType()),
+                $options
+            );
+
+            // for each field in the configuration form, we add a multistore checkbox
+            $fieldName = 'multistore_' . $child->getName();
```

Comment:
```suggestion
            $fieldName = 'multistore_' . $child->getName();
```

Could you set `'multistore_' ` in a constant ?